### ISSUE TITLE
Phase Out 32-bit Binaries

### ISSUE THREAD

**Issue by MikuAuahDark** [love2d/love#1889]
Currently LOVE still provides 32-bit builds in Windows and Android. On other platforms, 32-bit binaries are no longer provided.

Some rationale to motivate this change:
* On macOS, LOVE already provides 64-bit macOS binary since 0.9.x releases and starting from macOS 10.15, 32-bit-only apps are no longer executable.
* iOS support in LOVE is 64-bit by default (I guess?)
* On Android, 64-bit binaries are provided starting from 11.2 and 32-bit Android are declining rapidly.
* On Windows, Windows 11 only provides 64-bit ISO which means users who are running Windows 11 always have 64-bit. Users who are running Windows 10 32-bit (with supported hardware) can't upgrade to Windows 11 unless they (re)install (to) 64-bit Windows first.
* On Linux (few examples):
  * Ubuntu phased out 32-bit support in 19.10.
  * ArchLinux phased out 32-bit support way earlier at 2017.

I don't think doing this in 12.0 is a good idea, so maybe in 13.0.

**Comment by MikuAuahDark** [comment]
Hmm, in Android there are possible potential I have to drop 32-bit x86 binary in Android when LOVE 12.0 comes out. android/ndk#1772.

**Comment by slime73** [comment]
I think I'm ok with dropping official 32 bit builds for Windows in love 12 unless there are major objections - less than 0.1% of Steam users have a 32 bit Windows OS if I'm reading their stats right, and that number will continue to drop over time.

**Comment by MikuAuahDark** [comment]
I think it should go on 13.0. I still don't want to remove ARMv7 for Android yet as it still has around 19% of the market shares.

**Comment by slime73** [comment]
That's fine for Android. For Windows I don't see a reason to keep the 32 bit build right now.

**Comment by slime73** [comment]
32 bit windows builds are gone from Github Actions now and are mentioned in the release notes, so I'll leave the remaining bit of the issue for later.

**Comment by lee2sman** [comment]
Hi, have been working with older laptops (2008 Eee PC), early 2010s windows machines. it would be great to have easy-to-use instructions for building 32bit versions of love if an appimage binary cannot be provided. for example, the last debian that ships love 32 bit binaries is bullseye, so i had to download that old distro (actually, i used the derivative distro antiX 21). but then `sudo apt install love` didn't actually result in a working binary.  for good measure also installed `libopenmpt-modplug1`. but still got error that `liblove-11.3.so` could not be found even when in same folder, so the solution was to move it to the system path: `sudo cp .libs/liblove-11.3.so /usr/lib/` and then love runs.

part of the goal is to have working Love on older computers, good for longevity/preservaion of old games and software, and good for keeping older computers relevant and alive and less apt to be thrown away if they're still usable for running and writing games and other programs.

**Comment by MikuAuahDark** [comment]
That'd be probably hard situation.

The easiest way to get 32-bit LOVE AppImage is to compile on 32-bit platforms. This is because https://github.com/love2d/love-appimage-source assumes no cross compiling. However that repository is fully untested on 32-bit platforms.

Since you mentioned 2010-era PC, that means it's likely lacking modern instruction sets. So other than making cross-compile work on that repository or making 32-bit host compilation to work, you also need to ensure correct build flags (e.g. limiting compilers to only emit SSE2) being correctly set and propagated. Failing to do so will result in 32-bit binary that cannot be run in those PCs.

Furthermore, those PCs is _unlikely_ support modern graphics API LOVE requires. This will result in software rendering. Unfortunately I don't think those PCs have the juice to even run a slightly moderate game due to software rendering taking most of the processing power. Take an example of Balatro. Balatro may run on even modest 64-bit PC (thanks to their slow-but-modern iGPU which is better than nothing) but may struggle in those devices due to software rendering + lack of processing power.

So in the end, it doesn't worth supporting 32-bit devices anymore.

**Comment by lee2sman** [comment]
I see. Thanks for that info.

As background I am using Love 11.3 on this old Eee PC. Running it with software rendering some programs can be quite slow (such as my Conways' Game of Life implementation I just tested), though of course turn-based roguelikes, mathematical patterns and puzzles all work fine as well as some other realtime graphics. 

Thanks to pgimeno and farvardin in this [forum thread](https://www.love2d.org/forums/viewtopic.php?p=234660#p234660) with some additional work I was able to have a usable 11.3 binary and was hoping for some *official* binary that I could point people with similar machines to, so they could continue to make and play games with their machines.

**Comment by MikuAuahDark** [comment]
Quick look at the patch, 4ab9a1ce8cbffb30a38476818ff5d3989196683f from #1916 is probably related.

**Comment by lee2sman** [comment]
Yes, that was the error and the way to fix it!, along with that weird error that i got that `liblove-11.3.so could not be found` so had to move it to my path.

Also, I am not discounting those Steam/Unity stats from way up above but just to provide a counterpoint: I really value Love and its developers and the community for a number of reasons, including that Love works on a variety of machines including lower end ones. It helps to reduce consumption of new computers and helps with longevity of software, which I think is valuable, and the slow careful development process, which is in contrast to something like Unity that changes their IDE often, and at the whims of their large (commercial/military/etc) client base.
